### PR TITLE
Vet/race fixes

### PR DESCRIPTION
**Pull request by jefferai**
This fixes all vet issues and all races. It also enables the race detector as part of the CI workflows.

### CODE REVIEW

File: internal/cmd/base/logging.go [hunk 1]
```diff
@@ -59,77 +61,141 @@ func ProcessLogLevelAndFormat(flagLogLevel, flagLogFormat, configLogLevel, confi
...
+		logger: new(atomic.Pointer[grpcLogFakerLogHolder]),
+	}
+	grpclog.SetLoggerV2(grpcLogFaker)
+}
+
 type GRPCLogFaker struct {
-	Logger hclog.Logger
-	Log    bool
+	logger *atomic.Pointer[grpcLogFakerLogHolder]
+	log    *atomic.Bool
```
Comment (johanbrandhorst) [inline internal/cmd/base/logging.go:80]: Maybe name this something like `loggingEnabled` while we're here? It was a bit confusing to me to see `logger` and `log` in the same struct.
```suggestion
	loggingEnabled    *atomic.Bool
```

File: internal/cmd/commands/server/listener_reload_test.go [hunk 2]
```diff
@@ -83,7 +83,6 @@ listener "tcp" {
 `
 
 func TestServer_ReloadListener(t *testing.T) {
-	t.Parallel()
```
Comment (johanbrandhorst) [inline internal/cmd/commands/server/listener_reload_test.go:86]: Could you add a comment with a warning that this test cannot be run in parallel (and why) so we don't reintroduce this accidentally?

File: internal/daemon/cluster/handlers/upstream_message_service_controller.go [hunk 3]
```diff
@@ -19,7 +19,11 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
-var upstreamMessageHandler sync.Map
+var upstreamMessageHandler *sync.Map
+
+func init() {
+	upstreamMessageHandler = new(sync.Map)
```
Comment (johanbrandhorst) [inline internal/daemon/cluster/handlers/upstream_message_service_controller.go:25]: Can't we do this in the declaration instead of during init?

**Review by johanbrandhorst — COMMENTED** [review]
Are we also enabling race testing to prevent regressions or new failures? This is great ⭐

**Comment by jefferai** [comment]
>Are we also enabling race testing to prevent regressions or new failures?

Yes, but I think I'll do that when I'm done with the rest of the linting, and enable that at the same time.

File: internal/cmd/base/logging.go (hunk 1 above)
Comment (jefferai) [reply internal/cmd/base/logging.go:80]: Done.

File: internal/daemon/cluster/handlers/upstream_message_service_controller.go (hunk 3 above)
Comment (jefferai) [reply internal/daemon/cluster/handlers/upstream_message_service_controller.go:25]: Ah yeah fair, fixed

File: internal/cmd/commands/server/listener_reload_test.go (hunk 2 above)
Comment (jefferai) [reply internal/cmd/commands/server/listener_reload_test.go]: That actually I will revert...it was left over from trying to figure out one of the data races.

**Review by jimlambrt — APPROVED** [review]
ty!

**Pull request merged**